github/github.go

```diff
@@ -235,6 +235,23 @@ func NewClient(httpClient *http.Client) *Client {
 	return c
 }
 
+// NewEnterpriseClient returns a new GitHub API client (same as NewClient),
+// except it allows injection of custom base and upload urls. These urls might
+// be the same, but for flexibility both can be set separately.
+func NewEnterpriseClient(httpClient *http.Client, baseURL string, uploadURL string) *Client {
+	if !strings.HasSuffix(baseURL, "/") {
+		baseURL = baseURL + "/"
+	}
+	if !strings.HasSuffix(uploadURL, "/") {
+		uploadURL = uploadURL + "/"
+	}
+
+	c := NewClient(httpClient)
+	c.BaseURL, _ = url.Parse(baseURL)
+	c.UploadURL, _ = url.Parse(uploadURL)
```

Comment:
Can you provide rationale for why you decided to ignore errors instead of returning them, as I did in the sample in https://github.com/google/go-github/issues/756#issuecomment-338877501?

I don't necessarily disagree with the decision, I just want it to be explicit so we can understand it better and agree on.

Reply:
My rationale wasn't super strong, I did this to remain consistent with `NewClient`. But thinking about it now, I don't completely agree with that because we are now allowing the passing in of params that could be malformed. I would have nothing against returning an error to the user if parsing fails. I think this would be friendlier and useful.

Thanks for your in depth review! @shurcooL :)